core/internal/cltest/cltest.go

```diff
@@ -1101,6 +1101,14 @@ func LegacyTransactionsFromGasPrices(gasPrices ...int64) []gas.Transaction {
 	return txs
 }
 
+func SpecialTypeLegacyTransactionsFromGasPrices(code gas.TxType, gasPrices ...int64) []gas.Transaction {
```

Comment:
nit/ Typically in Go the idiom is to have the modifier as a trailing suffix so that they sort next to one another in the godocs. Also, the type name is generally used for something like this, and is actually shorter here: `TxType`.
```suggestion
func LegacyTransactionsFromGasPricesTxType(code gas.TxType, gasPrices ...int64) []gas.Transaction {
```

Reply:
It also looks like you could update the implementations of the normal tx versions to call through these and pass `0x0`/`0x2`.

Reply:
I'm not too keen on this change because it will increase verbosity by adding an extra argument to almost 60 test cases. The majority of the tests only use the `0x0/0x2` type.

Reply:
For example:
```diff
func LegacyTransactionsFromGasPrices(gasPrices ...int64) []gas.Transaction {
	return TransactionsFromGasPricesTxType(0x0, gasPrices...)
}
```